src/getter.js

```diff
@@ -9,6 +11,19 @@ import { isKeyPath } from './key-path'
  */
 const identity = (x) => x
 
+class GetterClass {
```

Comment:
Ok -- `addOptionsToGetter` was by ref. Which would modify it globally. We could return a cloned copy with options added instead. Thoughts?